realm/realm-library/src/testUtils/java/io/realm/TestHelper.java

```diff
@@ -1258,11 +1258,22 @@ public static void populateLinkedDataSet(Realm realm) {
      *
      * If the build does not support Sync, this method will do nothing
      */
-    public static void waitForNetworkThreadExecutorToFinish() {
+    private static final Field networkPoolExecutorField;
+    static {
         try {
             Class syncManager = Class.forName("io.realm.SyncManager");
-            Field field = syncManager.getDeclaredField("NETWORK_POOL_EXECUTOR");
-            ThreadPoolExecutor pool = (ThreadPoolExecutor) field.get(null);
+            networkPoolExecutorField = syncManager.getDeclaredField("NETWORK_POOL_EXECUTOR");
+        } catch (ClassNotFoundException | NoSuchFieldException e) {
+            networkPoolExecutorField = null;
```

Comment:
this won't compile :) 
You should throw an assertion error like we do in `SyncTestUtils`. 
Btw I think `waitForNetworkThreadExecutorToFinish` should be on the `objectServer` flavour i.e inside `SyncTestUtils` not `TestHelper` which is a helper for the base flavour mainly

Reply:
Yes, I agree it would have been better if this could be in `SyncTestUtils`, but I cannot move it there since this method is also called from `RunInLooperThread` which is also used by our base variant tests which do not have any Sync classes compiled in.